text/0000-spec-requirements.md

```diff
@@ -0,0 +1,53 @@
+# Spec Requirements
+
+## Test writing
+
+### Tests are written as framework-agnostic
+* Should support unit and end-to-end tests
+* JavaScript executed on the DOM is separate from the assertions
+* Tests accept `window` and optional DOM fragment
+    * DOM and window creation should be separate from the tests
+    * Tests should not need to know what framework the HTML was created from
+    * `window` should be used to derive `document`
+    * `window` is required within the tests to perform certain functionality in the browser (e.g. input/label connection)
+* individual tests or test group setup is dictated by the runner (unit setup vs integration setup)
+    * allow the tests to receive a setup (e.g. before/beforeAll)
+
+### Tests are written to be portable
+* Assertions do not have any dependencies outside of assertion definition (e.g. If Jest Assertions are used, the the Jest test framework does not also have to be used)
+    * Both Chai or Jest assertion libraries fit this requirement
+* Tests are grouped by functionality (e.g. opening an accordion (group) contains separate tests for keyboard users, mouse users, and screen reader users)
```

Comment:
This feels heavily tied to an implementation in the sense that there are multiple ways to express the relationship between tests. What I mean by this is that if we have the following relationship:

```
- Component
  - Test A
  - Test B
  - Test C
    - Test D
```

That we need a way to express a requirement around what tests a test may depend on. In this case, this could be that `Test D` depends on `Test C`. We could also express the ordering of the tests as an implicit dependency, meaning that `Test C` depends on `Test A` and `Test B` to be true, `Test B` depends on `Test A` to be true.

All this is to say that we should express the underlying requirement instead of one manner in which we can achieve that requirement. To reduce this down, we could say that there needs to be a mechanism by which we can express dependencies for tests. Dependencies can mean that these are tests that must have already passed before this test could be run. It could take on other meanings, as well, and that feels very appropriate as a part of the spec requirement document.

Reply:
This isn't so much about tests depending on one another, but more that _tests grouped together to test a specific feature_ of the component would be a feature which would help in inheritance. 

For instance, in dropdown, when using a disabled link as one of the `options`, we would want to specifically grab the group of tests that test for our expectations of links which are disabled. So, in this case `Tests are grouped by functionality` would be referencing a group of tests for disabled links.